lib/compress.js

```diff
@@ -1053,7 +1053,7 @@ merge(Compressor.prototype, {
             var val = {};
             for (var i = this.properties.length; --i >= 0;) {
                 var prop = this.properties[i];
-                if (prop instanceof AST_ObjectKeyVal) {
+                if (typeof prop.key === "string") {
```

Comment:
>  I think try-catch-ing the ev(prop.value, compressor); here would make sense as I want to skip over those failures.

No, you must not catch those exceptions - they indicate a side effect and must be allowed to be thrown.

ev() always has to be evaluated on the prop.value regardless of the key.

For ES5/master the key is always constant AFAIK, so ev() does not have to be run on the key. But in harmony (if supported) then ev() must always be run on AST_Node prop.key values in case it throws.